etna/transforms/timestamp/event.py

```diff
@@ -0,0 +1,155 @@
+
+        super().__init__(required_features=[in_column])
+        self.in_column = in_column
+        self.out_column = out_column
+        self.n_pre = n_pre
+        self.n_post = n_post
+        self.mode = mode
+        self.in_column_regressor: Optional[bool] = None
+
+    def fit(self, ts: TSDataset) -> "EventTransform":
+        """Fit the transform."""
+        self.in_column_regressor = self.in_column in ts.regressors
+        super().fit(ts)
+        return self
+
+    def _fit(self, df: pd.DataFrame):
+        """Fit method does nothing and is kept for compatibility.
+
+        Parameters
+        ----------
+        df:
+            dataframe with data.
+        """
+        pass
+
+    def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Mark days before and after event.
+
+        Parameters
+        ----------
+        df:
+            dataframe with data to transform.
+
+        Returns
+        -------
+        :
+            transformed dataframe
+
+        """
+        indexes = df.copy()
+        indexes[:] = np.repeat((np.arange(len(indexes)) + 1).reshape(-1, 1), len(indexes.columns), axis=1)
+
+        prev = df.copy()
+        prev.mask(prev != 1, None, inplace=True)
+        prev = prev * indexes
+        prev = prev.bfill().fillna(indexes)
+        prev = prev - indexes
+        if self.mode == 'binary':
+            prev.mask((prev >= 1) & (prev <= self.n_pre), 1, inplace=True)
+            prev.mask(prev > self.n_pre, 0, inplace=True)
+        else:
+            prev.mask(prev > self.n_pre, 0, inplace=True)
+            prev.mask((prev >= 1) & (prev <= self.n_pre), 1 / prev, inplace=True)
+        prev.rename(columns={self.in_column: f'{self.out_column}_prev'}, inplace=True, level="feature")
+
+        post = df.copy()
+        post.mask(post != 1, None, inplace=True)
+        post = post * indexes
+        post = post.ffill().fillna(indexes)
+        post = indexes - post
+        if self.mode == 'binary':
+            post.mask((post >= 1) & (post <= self.n_post), 1, inplace=True)
+            post.mask(post > self.n_post, 0, inplace=True)
+        else:
+            post.mask(post > self.n_post, 0, inplace=True)
+            post.mask((post >= 1) & (post <= self.n_post), 1 / post, inplace=True)
+        post.rename(columns={self.in_column: f'{self.out_column}_post'}, inplace=True, level="feature")
+
+        df = pd.concat([df, prev, post], axis=1)
+
+        return df
+
+    def get_regressors_info(self) -> List[str]:
+        """Return the list with regressors created by the transform."""
+        if self.in_column_regressor is None:
+            raise ValueError("Fit the transform to get the correct regressors info!")
+        return [self.out_column + '_pre', self.out_column + '_post'] if self.in_column_regressor else []
+
+    def _params_to_tune(self) -> Dict[str, BaseDistribution]:
```

Comment:
Why it private? The original method is public [see](https://github.com/etna-team/etna/blob/3489d1de4dfad861a4f64c8e0a9aa2f48ed76da3/etna/transforms/base.py#L167)